signer/src/transaction_coordinator.rs

```diff
@@ -346,13 +342,28 @@ where
                 Ok(key) => key,
                 Err(error) => {
                     tracing::error!(%error, "failed to coordinate DKG; using existing aggregate key");
-                    registry_aggregate_key
+                    registry_signer_set_info
+                        .as_ref()
+                        .map(|info| info.aggregate_key)
                         .ok_or(Error::MissingAggregateKey(*bitcoin_chain_tip.block_hash))?
                 }
             }
         } else {
-            registry_aggregate_key
-                .ok_or(Error::MissingAggregateKey(*bitcoin_chain_tip.block_hash))?
+            // If we do not have signer set info in the registry, then we
+            // are in the bootstrap phase. Our latest DKG shares may be
+            // 'Unverified', but if we are here then they are not 'Failed',
+            // since we will coordinate DKG if our last DKG shares are
+            // 'Failed'.
```

Comment:
IIUC we can have a race here, given that between `coordinate_dkg` getting the shares count and the `get_latest_encrypted_dkg_shares` below we could have the block observer marking the latest shares as failed, so this may not be always true. But I guess it's fine regardless, since then in `check_and_submit_rotate_key_transaction` we would err with `DkgVerificationFailed` and skip the tenure. Maybe consider updating the comment?